src/web/src/app/space/detail/component.ts

```diff
@@ -0,0 +1,277 @@
+		this.getLinks(this.linksPageSize, 0);
+	}
+
+	onUserDelBtn(user: SpaceUser){
+		this.space.delUserById(user.userId).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		).subscribe(res => {});
+		this.usersPaginator.firstPage();
+		this.getUsers(this.usersPageSize, 0);
+	}
+
+	getUserNameById(userId: string) {
+		let name;
+		this.user.getById(userId)
+		.subscribe(res => {
+			name = res.name;
+		});
+		return name;
+	}
+
+	getUserEmailById(userId: string) {
+		let email;
+		this.user.getById(userId)
+		.subscribe(res => {
+			email = res.email;
+		});
+		return email;
+	}
+
+	getUsernameById(userId: string) {
+		let username;
+		this.user.getById(userId)
+		.subscribe(res => {
+			username = res.username;
+		});
+		return username;
+	}
+
+	private _initInvitationForm() {
+		this.invitationForm = this.fb.group({
+			username: ['', [
+				Validators.required,
+				Validators.pattern('[a-z0-9_]*'),
+			]],
+			role: ['', [
+				Validators.required
+			],],
+		});
+	}
+	
+	onInvitationAdd() {
+		this.isInvitationFormHidden = false;
+	}
+
+	onInvitationFormCloseBtn() {
+		this.isInvitationFormHidden = true;
+		this.invitationForm.reset();
+	}
+
+	onSubmitInvitation(data): void {
+		if (this.invitationForm.invalid) {
+			return;
+		}
+		let userId;
+		this.user.getByUsername(data.value.username)
+		.subscribe(res => {
+			userId = res.id;
+		});
+	
+		this.space.createInvitation(
+			this.currentSpaceId,
+			userId,
+			data.value.role,
+			this.currentUserId
+		).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		).subscribe(res => {});
+			
+		this.invitationForm.reset();
+		this.isInvitationFormHidden = true;
+		this.invitationsPaginator.firstPage();
+		this.getInvitations(this.invitationsPageSize, 0);
+	}
+
+	private _initLinkForm() {
+		this.linkForm = this.fb.group({
+			name: ['', [
+				Validators.required,
+				Validators.pattern('[a-z0-9_]*'),
+			]],
+			expiredAt: [null, [Validators.required]],
+		});
+	}
+
+	onLinkAdd() {
+		this.isLinkFormHidden = false;
+	}
+
+	onLinkFormCloseBtn() {
+		this.isLinkFormHidden = true;
+		this.linkForm.reset();
+	}
+
+	onSubmitLink(data): void {
+		if (this.linkForm.invalid) {
+			return;
+		}
+		this.space.createLink(
+			this.currentSpaceId,
+			this.currentUserId,
+			data.value.name,
+			data.value.expiredAt
+		).pipe(
+			src => this.requestWatcher.WatchFor(src)
+		).subscribe(res => {});
+
+		this.linkForm.reset();
+		this.isLinkFormHidden = true;
+		this.linksPaginator.firstPage();
+		this.getLinks(this.linksPageSize, 0);
+	}
+
+	savePageSizeToLocalStorage(id: string, pageEvent: PageEvent) {
+		localStorage.setItem(id+'PageSize', pageEvent.pageSize.toString());
+		switch(id){
+			case "invitationsPaginatorSpaceDetail":
+				this.invitationsPageSize = pageEvent.pageSize;
+				break;
+			case "linksPaginatorSpaceDetail":
+				this.linksPageSize = pageEvent.pageSize;
+				break;
+			case "usersPaginatorSpaceDetail":
+				this.usersPageSize = pageEvent.pageSize;
+				break;
+		}
+	}
+}
```

Comment:
Нужно проверить что изменение значения в this.pageSize не приводит к двойному запросу списков с сервера. Т.е. пагинатор может проверить что оригинальное значение изменилось и не сверив его со своим текущим, обновить страницу. 

Если такое происходит нужно будет дублировать this.pageSize и изменять дубль, а не оригинал.

```suggestion
	savePageSize(id: string, limit: number) {
		if (this.pageSize[id] == limit)
			return;
		
		this.pageSize[id] = limit;
		localStorage.setItem('spaceDetailPageSize', JSON.stringify(this.pageSize));
	}
}
```

Reply:
Сделал, но условие не понял.

Reply:
this.pageSize это у нас структура с полями invitations, links, users. В эту функцию в id мы передаём имя одного из поля структуры. Получаем значение поля и сравниваем с новым значением. Если значение не изменилось, то выходим, т.к. не нужно в localStorage обновлять.

А ты проверил не вызывается ли запрос списка два раза если исходный this.pageSize обновлять?

Reply:
Ты пишешь:

"А ты проверил не вызывается ли запрос списка два раза если исходный this.pageSize обновлять?"

Ты имеешь в виду двойной запрос, к примеру, getInvitations()?

Reply:
1. Добавить вывод в лог какого нибудь глобального счётчика в самом начале функции например getInvitations. 
2. Перейти в пагинаторе на вторую страницу.
3. Изменить кол-во отображаемых строк в пагинаторе.
4. Посмотреть сколько счётчик насчитал вызовов getInvitations.

Reply:
Проверил. Не наблюдаю двойных вызовов, только одинарные. При каждом переходе на новую страницу либо при смене кол-ва отображаемых элементов в пагинаторе - каждый раз срабатывает getInvitations(), но по одному разу.

Reply:
Спасибо что проверил!